PyDSS/get_snapshot_timepoints.py

```diff
@@ -0,0 +1,82 @@
+"""Logic to determine snapshot time point by mode"""
+
+import logging
+
+from PyDSS.common import SnapshotTimePointSelectionMode
+from PyDSS.utils.simulation_utils import create_loadshape_pmult_dataframe_for_simulation
+from PyDSS.reports.reports import logger
+
+import pandas as pd
+import opendssdirect as dss
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_snapshot_timepoint(options, mode: SnapshotTimePointSelectionMode):
+    pv_generation_hours = {'start_time': '8:00', 'end_time': '17:00'}
+    pv_systems = dss.PVsystems.AllNames()
+    if not pv_systems:
+        logger.info("No PVSystems are present.")
+        if mode != SnapshotTimePointSelectionMode.MAX_LOAD:
+            mode = SnapshotTimePointSelectionMode.MAX_LOAD
+            logger.info("Changed mode to %s", SnapshotTimePointSelectionMode.MAX_LOAD.value)
+    pv_profiles = {}
+    index = None
+    for pv_name in pv_systems:
+        dss.PVsystems.Name(pv_name)
+        pmpp = float(dss.Properties.Value('Pmpp'))
+        profile_name = dss.Properties.Value('yearly')
+        dss.LoadShape.Name(profile_name)
+        df = create_loadshape_pmult_dataframe_for_simulation(options) * pmpp
```

Comment:
@ann-sherin This code is causing lots of delay. We only need to create the dataframe once. We can create one dataframe before entering the for loop and then copy it each time. As it stands, we are calling this function thousands of times across PVSystems and Loads. The get_snapshot_timepoint function is taking up to 3 minutes with year-long time series. Can you fix it?